pkg/executor/scraper/factory/factory.go

```diff
@@ -19,36 +18,47 @@ import (
 	"github.com/kubeshop/testkube/pkg/ui"
 )
 
-func Scrape(ctx context.Context, dirs []string, execution testkube.Execution, params envs.Params) (err error) {
-	output.PrintLog(fmt.Sprintf("%s Extracting artifacts from %s using Filesystem Extractor", ui.IconCheckMark, dirs))
+type ExtractorType string
+type UploaderType string
 
-	extractor := scraper.NewFilesystemExtractor(dirs, filesystem.NewOSFileSystem())
+const (
+	RecursiveFilesystemExtractor ExtractorType = "RecursiveFilesystemExtractor"
+	ArchiveFilesystemExtractor   ExtractorType = "ArchiveFilesystemExtractor"
+	MinIOUploader                UploaderType  = "MinIOUploader"
+	CloudUploader                UploaderType  = "CloudUploader"
+)
+
+func GetScraper(ctx context.Context, params envs.Params, extractorType ExtractorType, uploaderType UploaderType) (scraper.Scraper, error) {
+	var extractor scraper.Extractor
+	switch extractorType {
+	case RecursiveFilesystemExtractor:
+		extractor = scraper.NewRecursiveFilesystemExtractor(filesystem.NewOSFileSystem())
```

Comment:
Should we have an option to do recursive scanning or not? Because in some cases you don't to retrieve sub folders with temp data or so?